src/index/IndexImpl.cpp

```diff
@@ -896,7 +896,8 @@ void IndexImpl::readConfiguration() {
   f >> configurationJson_;
   if (configurationJson_.find("git-hash") != configurationJson_.end()) {
     LOG(INFO) << "The git hash used to build this index was "
-              << std::string(configurationJson_["git-hash"]).substr(0, 6)
+              << configurationJson_["git-hash"].get<std::string_view>().substr(
+                     0, 6)
```

Comment:
If you don't use the short hash (because of compatibility I assume, what's the point of storing it?)
Maybe it is easier to implement a function in the compilation Info that takes either a string_view or a json object and gives you the short hash.